internal/olm/operator/internal/registry_pod_test.go

```diff
@@ -0,0 +1,169 @@
+			})
+
+			It("should create registry pod successfully", func() {
+				err := rp.Create(context.Background())
+
+				Expect(err).To(BeNil())
+			})
+
+			It("check pod status should return successfully when pod check is true", func() {
+				mockGoodPodCheck := wait.ConditionFunc(func() (done bool, err error) {
+					return true, nil
+				})
+
+				err := rp.checkPodStatus(context.Background(), mockGoodPodCheck)
+
+				Expect(err).To(BeNil())
+			})
+
+		})
+
+		Context("with invalid registry pod values", func() {
+
+			It("should error when bundle image is not provided", func() {
+				expectedErr := "bundle image cannot be empty"
+
+				_, err := NewRegistryPod(newFakeClient(), "/database/index.db",
+					"", "default")
+
+				Expect(err).NotTo(BeNil())
+				Expect(err.Error()).Should(ContainSubstring(expectedErr))
+			})
+
+			It("should not create a registry pod when namespace is not provided", func() {
+				expectedErr := "namespace cannot be empty"
+
+				_, err := NewRegistryPod(newFakeClient(), "/database/index.db",
+					"quay.io/example/example-operator-bundle:0.2.0", "")
+
+				Expect(err).NotTo(BeNil())
+				Expect(err.Error()).Should(ContainSubstring(expectedErr))
+			})
+
+			It("should not create a registry pod when database path is not provided", func() {
+				expectedErr := "registry database path cannot be empty"
+
+				_, err := NewRegistryPod(newFakeClient(), "",
+					"quay.io/example/example-operator-bundle:0.2.0", "default")
+
+				Expect(err).NotTo(BeNil())
+				Expect(err.Error()).Should(ContainSubstring(expectedErr))
+			})
+
+			It("should error when invalid bundle image is provided", func() {
+				expectedErr := "invalid bundle image name"
+
+				_, err := NewRegistryPod(newFakeClient(), "/database/index.db",
+					"non/existent/image", "default")
+
+				Expect(err).NotTo(BeNil())
+				Expect(err.Error()).Should(ContainSubstring(expectedErr))
+			})
+
+			It("should not accept any other bundle add mode other than semver or replaces", func() {
+				expectedErr := "invalid bundle mode"
+
+				rp, _ := NewRegistryPod(newFakeClient(), "/database/index.db",
+					"quay.io/example/example-operator-bundle:0.2.0", "default")
+				rp.BundleAddMode = "invalid"
+
+				err := rp.validate()
+				Expect(err).NotTo(BeNil())
+				Expect(err.Error()).Should(ContainSubstring(expectedErr))
+			})
+
+			It("check pod status should return error when pod check is false and context is done", func() {
+				rp, _ := NewRegistryPod(newFakeClient(), "/database/index.db",
+					"quay.io/example/example-operator-bundle:0.2.0", "default")
+
+				mockBadPodCheck := wait.ConditionFunc(func() (done bool, err error) {
+					return false, fmt.Errorf("error waiting for registry pod")
+				})
+
+				expectedErr := "error waiting for registry pod"
+				// create a new context with a deadline of 400 milliseconds
+				ctx, cancel := context.WithTimeout(context.Background(), 400*time.Millisecond)
+				defer cancel()
```

Comment:
This may lead to flaky unit tests on faster machines. You may be able to just call `cancel()` after creation to get the same effect:
```suggestion
				ctx, cancel := context.WithTimeout(context.Background(), 1*time.Millisecond)
				cancel()
```

Reply:
Very good point @estroz, fixed it as per your suggestion.